File: src/erl/file_system.erl
-module(file_system).
-export([start/0]).

start() ->
    {ok, CurrentDirectory} = file:get_cwd(),
    FolderPath = CurrentDirectory ++ "/../files/",

    {ok, ChildItems} = file:list_dir(FolderPath),

    Folders = lists:filter(fun(ChildItem) -> filelib:is_dir(FolderPath ++ ChildItem) end, ChildItems),
    
    Files = lists:filter(fun(ChildItem) -> 
        filelib:is_file(FolderPath ++ ChildItem) and (not filelib:is_dir(FolderPath ++ ChildItem)) 
    end, ChildItems),

    lists:foreach(fun(Folder) -> io:format("/~s~n", [Folder]) end, Folders),
    lists:foreach(fun(File) -> io:format("~s,~wB~n", [File, filelib:file_size(FolderPath ++ File)]) end, Files),

    erlang:halt().

File: src/go/file_system.go
package main

import (
    "fmt"
	"os"
	"io/ioutil"
)

func main() {
	var folders []os.FileInfo
	var files []os.FileInfo

	childItems, _ := ioutil.ReadDir("../files")
	
	for _, childItem := range childItems {
		if childItem.IsDir(){
			folders = append(folders, childItem)
		}else{
			files = append(files, childItem)
		}
	}

	for _, folder := range folders {
		fmt.Println("/" + folder.Name())
	}

	for _, file := range files {
		fmt.Printf("%s,%dB\n", file.Name(), file.Size())
	}
}